tests/test_scheduler/test_unit/test_base.py

```diff
@@ -108,8 +108,12 @@ def test_handle_cuda(self,mocked_init):
                     component._handle_cuda(unit)
             else:
                 component._handle_cuda(unit)
-                self.assertEqual(unit['description']['environment']['CUDA_VISIBLE_DEVICES'],
-                                 result)
+                unit_env = unit['description']['environment']
+                if result == 'KeyError':
+                    with pytest.raises(KeyError):
+                        self.assertIsNone(unit_env['CUDA_VISIBLE_DEVICES'])
```

Comment:
What is this line supposed to test? Can you provide some more information?

Reply:
accessing the key `'CUDA_VISIBLE_DEVICES'` will raise an exception. Initially I did `_ = unit_env['CUDA_VISIBLE_DEVICES']`, but flake gave a warning, that's why such construction was chosen